src/ledger/result_writer.zig

```diff
@@ -110,58 +110,143 @@ pub const LedgerResultWriter = struct {
     /// agave: set_duplicate_confirmed_slots_and_hashes
     pub fn setDuplicateConfirmedSlotsAndHashes(
         self: *Self,
-        duplicate_confirmed_slot_hashes: []struct { Slot, Hash },
+        duplicate_confirmed_slot_hashes: []const struct { Slot, Hash },
     ) !void {
-        var write_batch = try self.db.initWriteBatch();
+        var setter = try self.setDuplicateConfirmedSlotsAndHashesIncremental();
+        defer setter.deinit();
+        errdefer setter.cancel();
+
         for (duplicate_confirmed_slot_hashes) |slot_hash| {
-            const slot, const frozen_hash = slot_hash;
-            const data = FrozenHashVersioned{ .current = FrozenHashStatus{
+            const slot, const hash = slot_hash;
+            try setter.addSlotAndHash(slot, hash);
+        }
+
+        try setter.commit();
+    }
+
+    /// Returns a struct which can be used to enact the same operation as `setDuplicateConfirmedSlotsAndHashes`, incrementally.
```

Comment:
Nit: Is this comment correct? or `setDuplicateConfirmedSlotsAndHashes` is meant to mean Agave's `set_duplicate_confirmed_slots_and_hashes`? Because if not, then it is confusing as Sig `setDuplicateConfirmedSlotsAndHashes` makes use of it and hence `setDuplicateConfirmedSlotsAndHashes` in Sig is incremental

Reply:
I'm not sure what you mean exactly, "incrementally" is referring to the fact that it doesn't take a slice parameter, and instead returns a struct that allows you to do the operation multiple times before committing it.

Reply:
> to enact the same operation as `setDuplicateConfirmedSlotsAndHashes`, incrementally

Unless I read this wrong, but the above alludes to  `setDuplicateConfirmedSlotsAndHashes` not being an incremental operation...and this comment says this method makes it possible to do same thing as `setDuplicateConfirmedSlotsAndHashes` but incrementally.

But it looks like `setDuplicateConfirmedSlotsAndHashes` is incremental because it's implementation depends on `setDuplicateConfirmedSlotsAndHashes`

Reply:
`setDuplicateConfirmedSlotsAndHashes` is a single function call, which takes a parameter with all the duplicate confirmed slothashes to set in one go. It is implemented in terms of its incremental counterpart, but it does not allow the caller to incrementally set the slothashes, ergo it is not itself incremental.

Reply:
You are right. I see where I got myself confused by the comments.